src/java.base/share/classes/sun/security/ssl/SSLLogger.java

```diff
@@ -74,40 +75,40 @@ public final class SSLLogger {
                 if (property.equals("help")) {
                     help();
                 }
-
                 logger = new SSLConsoleLogger("javax.net.ssl", p);
             }
             isOn = true;
+            // log almost everything for the "ssl" value.
+            // anything else specified with "ssl" in the property value implies
+            // a subcomponent value is to be logged.
+            sslOn = property.equals("ssl");
         } else {
             property = null;
             logger = null;
             isOn = false;
+            sslOn = false;
         }
     }
 
     private static void help() {
         System.err.println();
-        System.err.println("help           print the help messages");
-        System.err.println("expand         expand debugging information");
```

Comment:
Why are we tossing `expand`?  I quickly looked at the CSR, but the rationale didn't really gel.  It wasn't mentioned in the actual bug report either.  I'll look more next week.

Anyway, if we decide this is in the best interest of the platform, then you should also delete the `useCompactFormat` and friends in the SSLConsoleLogger.

Reply:
good point. I think it might have been wiped away in earlier iterations. I've added it back and added test coverage for this option. The separator logic for such options isn't specified which makes coding more tricky. I've tried to keep the syntax parsing flexible.

Reply:
Maybe add a bit about what `expand` does?  

> expand       expanded (less compact) output format